src/frontend/screens/Settings/ProjectSettings/DeviceName.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react';
+import {useForm} from 'react-hook-form';
+import {Alert, Keyboard, StyleSheet} from 'react-native';
+import {NativeNavigationComponent} from '../../../sharedTypes';
+import {defineMessages, useIntl} from 'react-intl';
+import {Text} from '../../../sharedComponents/Text';
+import {TouchableWithoutFeedback} from 'react-native-gesture-handler';
+import {BLACK, MEDIUM_GREY, RED} from '../../../lib/styles';
+import {
+  useDeviceInfo,
+  useEditDeviceInfo,
+} from '../../../hooks/server/deviceInfo';
+import {Loading} from '../../../sharedComponents/Loading';
+import {HeaderRightDeviceName} from './HeaderRightDeviceName';
+import {HookFormTextInput} from '../../../sharedComponents/HookFormTextInput';
+
+const m = defineMessages({
+  title: {
+    id: 'Screens.Settings.ProjectSettings.DeviceName.title',
+    defaultMessage: 'Device Name',
+  },
+  yourDevice: {
+    id: 'Screens.Settings.ProjectSettings.DeviceName.yourDevice',
+    defaultMessage: 'Your Device Name',
+  },
+  editDevice: {
+    id: 'Screens.Settings.ProjectSettings.DeviceName.editDevice',
+    defaultMessage: 'Edit Device Name',
+  },
+  discardChanges: {
+    id: 'Screens.Settings.ProjectSettings.DeviceName.discardChanges',
+    defaultMessage: 'Discard Changes',
+  },
+  continueEditting: {
+    id: 'Screens.Settings.ProjectSettings.DeviceName.continueEditting',
+    defaultMessage: 'Continue Editting',
+  },
+});
+
+type FormData = {
+  name: string;
+};
+
+export const DeviceName: NativeNavigationComponent<'DeviceName'> = ({
+  navigation,
+}) => {
+  const [isEditting, setIsEditting] = React.useState(false);
+  const deviceInfo = useDeviceInfo();
+  const {mutate, isPending, variables: newDeviceName} = useEditDeviceInfo();
+  const {formatMessage: t} = useIntl();
+  const {
+    control,
+    handleSubmit,
+    watch,
+    formState: {errors},
+  } = useForm<FormData>({defaultValues: {name: ''}});
+
+  React.useEffect(() => {
+    navigation.setOptions({
+      headerRight: () => (
+        <HeaderRightDeviceName
+          handleSubmit={handleSubmit(val => {
+            mutate(val.name);
+            setIsEditting(false);
+          })}
+          setIsEditting={setIsEditting}
+          isEditting={isEditting}
+        />
+      ),
+    });
+  }, [navigation, isEditting, handleSubmit, setIsEditting]);
```

Comment:
```suggestion
  }, [navigation, isEditting, handleSubmit, setIsEditting, mutate]);
```

Can you fix #135 - bugs from something like this are really hard to track down.